pilot/model/config.go

```diff
@@ -433,6 +463,12 @@ func SortRouteRules(rules []Config) {
 }
 
 func (store *istioConfigStore) RouteRules(instances []*ServiceInstance, destination string) []Config {
```

Comment:
We can rename this as RouteRulesV1Alpha1.. in fact, along the lines of @ZackButcher 's PR, I would do another PR that renamed all these functions to v1alpha1routerules.

Or, we could have a simple if-else inside the RouteRules. Based on some static settings in pilot (v1/v2 api), we pick the right variant. This way, function names remain constant?